src/helper/serializer.ts

```diff
@@ -16,6 +16,8 @@ const serializeAttrs = (attrs: HtmlTagDescriptor['attrs']): string => {
   for (const key in attrs) {
     if (typeof attrs[key] === 'boolean') {
       res += attrs[key] ? ` ${key}` : ``
+    } else if (attrs[key] === undefined) {
```

Comment:
The logic to adjust type attr should not be on the serializer helper but directly on attributes generations.

Reply:
Yes, this is a just quick fix. This can be done in a more elegant way. But the serializer here doesn’t check the type of attr and can cause error when type is not string, for example, undefined.

Reply:
Yes you right, I will rewrite all the attributes part to ensure the user attributes override the generated one.

Reply:
The serializer I use is the same as vite so I can't make this change here. Between the two `injectTo` I don't use the same logic. So I will remove all attrs with undefined values before tags html gen.